src/Compilation.zig

```diff
@@ -3638,6 +3638,15 @@ pub fn generateBuiltinZigSource(comp: *Compilation, allocator: *Allocator) Alloc
         std.zig.fmtId(@tagName(comp.bin_file.options.machine_code_model)),
     });
 
+    const wasi_exec_model_fmt = if (comp.bin_file.options.wasi_exec_model) |model|
+        std.zig.fmtId(@tagName(model))
+    else
+        std.zig.fmtId(@tagName(std.builtin.WasiExecModel.command));
+    try buffer.writer().print(
+        \\pub const wasi_exec_model = std.builtin.WasiExecModel.{};
+        \\
+    , .{wasi_exec_model_fmt});
```

Comment:
FYI: If I put this into a single statement, 

```zig
    try buffer.writer().print(
        \\pub const wasi_exec_model = std.builtin.WasiExecModel.{};
        \\
    , .{if (comp.bin_file.options.wasi_exec_model) |model|
        std.zig.fmtId(@tagName(model))
    else
        std.zig.fmtId(@tagName(std.builtin.WasiExecModel.command))});
```

I got some weird error around LLVM: 
```
make install
[ 77%] Built target embedded_softfloat
[ 78%] Built target opt_c_util
[ 83%] Built target zigcpp
[ 97%] Built target zigstage1
[ 98%] Built target zig0
[ 99%] Building self-hosted component /home/mathetake/zig/build/zig1.o
broken LLVM module found: Instruction does not dominate all uses!
  %369 = getelementptr inbounds %"Compilation.struct:3644:8", %"Compilation.struct:3644:8"* %39, i32 0, i32 0, !dbg !75434
  call fastcc void @std.zig.fmt.fmtId(%"std.fmt.Formatter(std.zig.fmt.formatId)"* sret(%"std.fmt.Formatter(std.zig.fmt.formatId)") %369, %"[]u8"* @13528), !dbg !75435
Instruction does not dominate all uses!
  %369 = getelementptr inbounds %"Compilation.struct:3644:8", %"Compilation.struct:3644:8"* %39, i32 0, i32 0, !dbg !75434
  %372 = bitcast %"std.fmt.Formatter(std.zig.fmt.formatId)"* %369 to i8*, !dbg !75435
Instruction does not dominate all uses!
  %369 = getelementptr inbounds %"Compilation.struct:3644:8", %"Compilation.struct:3644:8"* %39, i32 0, i32 0, !dbg !75434
  %373 = bitcast %"std.fmt.Formatter(std.zig.fmt.formatId)"* %369 to i8*, !dbg !75435

This is a bug in the Zig compiler.
Aborted (core dumped)
make[2]: *** [CMakeFiles/zig.dir/build.make:328: zig1.o] Error 134
make[1]: *** [CMakeFiles/Makefile2:148: CMakeFiles/zig.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

Reply:
I believe this is because it's not enough to just add an enum to `std.builtin`. You need to make more tweaks to stage1 in C++. For starters, we need to have the exact same enum shared between both stage1 and stage2 which means you need add a corresponding enum to `src/stage1/stage1.h`. Then, you'll need to tweak logic for actually marshalling the data between stage1 and stage2 - have a look how this is for `CodeModel` or some other builtin structs.

If you get stuck, I'll be happy to help you out but will probably take a look at this in more detail tomorrow. Alternatively, we can also get @andrewrk involved if he's got a minute.

Reply:
Ah I see. Thanks for clarifying!

Reply:
so yeah I found that this also applies to other existing builtin types: 

```
const std = @import("std");

pub fn main() void {
    std.debug.print("test {}", .{if (@rem(std.time.timestamp(), 2) == 0)
        std.zig.fmtId(@tagName(std.builtin.CodeModel.tiny))
    else
        std.zig.fmtId(@tagName(std.builtin.CodeModel.default))});
}
```

HEAD and 0.8.0 fails to compile this code with the same error like this:

```
zig build-exe main.zig
broken LLVM module found: Instruction does not dominate all uses!d.debug.ModuleDebug...
  %2 = getelementptr inbounds %"struct:4:33", %"struct:4:33"* %0, i32 0, i32 0, !dbg !16731
  call fastcc void @std.zig.fmt.fmtId(%"std.fmt.Formatter(std.zig.fmt.formatId)"* sret(%"std.fmt.Formatter(std.zig.fmt.formatId)") %2, %"[]u8.18"* @390), !dbg !16733
Instruction does not dominate all uses!
  %2 = getelementptr inbounds %"struct:4:33", %"struct:4:33"* %0, i32 0, i32 0, !dbg !16731
  %5 = bitcast %"std.fmt.Formatter(std.zig.fmt.formatId)"* %2 to i8*, !dbg !16733
Instruction does not dominate all uses!
  %2 = getelementptr inbounds %"struct:4:33", %"struct:4:33"* %0, i32 0, i32 0, !dbg !16731
  %6 = bitcast %"std.fmt.Formatter(std.zig.fmt.formatId)"* %2 to i8*, !dbg !16733

This is a bug in the Zig compiler.thread 3208559 panic:
Unable to dump stack trace: debug info stripped
zsh: abort (core dumped)  zig build-exe main.zig
```

and the same, if we split the line to two statements. the error disappears.

Reply:
fyi: I tweaked the C++ and around stage1.zig and Compilation.zig following `CodeModel`, but the error has never gone.

Reply:
Luckily the code with split statements just works and successfully produced Wasi Reactors!